authors/apps/friends/views.py

```diff
@@ -0,0 +1,132 @@
+        """
+        Checks if the user associated with the username passed exists
+        If true, check for the user sending the request
+        If same user, returned error message
+        Else create the user relationship
+        Finally return the serialized object of the followed user
+        """
+        try:
+            followed = get_user_model().objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise Http404()
+
+        follower = get_user_model().objects.get(pk=request.user.id)
+
+        if followed == follower:
+            return Response(
+                {"message": "You cannot follow yourself"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        user_followed, created_now = Friend.objects.get_or_create(
+            user_from=follower, user_to=followed)
+        if created_now is False:
+            return Response(
+                {"message": "you already follow this user"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        serializer = FollowUnfollowSerializer(followed)
+        user_details = serializer.data
+        return Response(user_details, status=status.HTTP_200_OK)
+
+    def delete(self, request, username, format=None):
+        """
+        Check if the user associated with the username passed exists
+        If true, check for the user sending the request
+        If same user, return an error message
+        Else delete the user relationship
+        Finally the serialized object of the unfollowed user is returned
+        """
+        try:
+            followed = get_user_model().objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise Http404()
+
+        follower = get_user_model().objects.get(pk=request.user.id)
+
+        if followed == follower:
+            return Response(
+                {"message": "You cannot unfollow yourself"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        relationship = Friend.objects.filter(
+            user_from=follower, user_to=followed)
+
+        if not relationship:
+            return Response(
+                {"message": "you don't follow this user"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        relationship.delete()
+        serializer = FollowUnfollowSerializer(followed)
+        user_details = serializer.data
+        return Response(user_details, status=status.HTTP_200_OK)
+
+
+class FollowersApiView(generics.ListAPIView):
+    """view for listing all followers of a particular user"""
+
+    def get_queryset(self):
+        """
+        check if the user associated with the username passed exists
+        if true return all the friend objects associated with the user
+        """
+        user = get_object_or_404(
+            get_user_model(), username=self.kwargs['username'])
+        return Friend.objects.select_related(
+            'user_to').filter(user_to=user.id).all()
```

Comment:
The select_related field here should be `user_from` because that's what you use in line 102